Tests/Unit/LoRaTools/TwinCollectionExtensionsTests.cs

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace LoRaWan.Tests.Unit.LoRaTools.Regions
+{
+    using Microsoft.Azure.Devices.Shared;
+    using Xunit;
+    using System;
+    using System.Globalization;
+    using global::LoRaTools.Utils;
+    using Microsoft.Extensions.Logging.Abstractions;
+    using Microsoft.Extensions.Logging;
+
+    public class TwinCollectionExtensionsTests
+    {
+        private readonly ILogger logger;
+
+        public TwinCollectionExtensionsTests()
+        {
+            this.logger = NullLogger.Instance;
+        }
+
+        [Fact]
+        public void SafeRead_Returns_Default_If_Value_Does_Not_Exist()
+        {
+            var tc = new TwinCollection();
+            var reader = new TwinCollectionReader(tc, this.logger);
+
+            Assert.False(reader.TryRead<string>("test", out var someString));
+            Assert.Null(someString);
+
+            Assert.False(reader.TryRead<ushort>("test", out var someUShort));
+            Assert.Equal(0, someUShort);
+
+            Assert.False(reader.TryRead<ushort?>("test", out var someUShortNullable));
+            Assert.Null(someUShortNullable);
+        }
+
+        [Theory]
+        [InlineData(true, null)]
+        [InlineData(false, null)]
+        [InlineData(true, "")]
+        public void ReadRequiredString_Throws_If_String_Is_Not_Configured(bool add, string val)
+        {
+            const string key = "test";
+            var tc = new TwinCollection();
+            var reader = new TwinCollectionReader(tc, this.logger);
+            if (add)
+                tc[key] = val;
+
+            Assert.Throws<InvalidOperationException>(() => reader.ReadRequiredString(key));
+        }
+
+        [Fact]
+        public void ReadRequiredString_Reads_Valid_String()
+        {
+            const string key = "test";
+            const string expectedValue = "expected";
+
+            var tc = CreateTwinCollectionReader(key, QuoteJsonString(expectedValue));
```

Comment:
[nit] maybe it would be a bit simpler to use a specific overload for strings instead of always needing to think about quoting the JSON strings:

```csharp
private TwinCollectionReader CreateTwinCollectionReader(string key, string someString) =>
            CreateTwinCollectionReader<string>(key, $"'{someString}'");
```

Reply:
aeaea698600306ca283c59cf31a9ae4fdc5e8268